src/ImageProcessor.js

```diff
@@ -0,0 +1,255 @@
+                async.eachSeries(
+                    // Collection
+                    req.ops,
+                    // Iteration
+                    (opItem, nextOp) => {
+                        var imageUUID = uuid();
+                        var targetFile = path.join(destDir, imageUUID + extension);
+                        var command = null;
+                        switch (String(opItem.op).toLowerCase()) {
+                            
+                            case 'orient':
+                                command = `convert "${sourceFile}" -auto-orient "${targetFile}"`;
+                                break;
+                                
+                            case 'resize':
+                                var qualityOpt = '';
+                                if (typeof opItem.quality == 'number') {
+                                    qualityOpt = '-quality ' + opItem.quality;
+                                }
+                                command = `convert "${sourceFile}" -auto-orient -resize ${opItem.width}x${opItem.height} ${qualityOpt} "${targetFile}"`;
+                                break;
+                                
+                        }
+                        
+                        if (command) {
+                            child_process.exec(command, (err, stdout, stderr) => {
+                                if (err) {
+                                    console.error('ImageMagick error', {
+                                        command, stdout, stderr, err
+                                    });
+                                    nextOp(err);
+                                }
+                                else {
+                                    processedImages.push({
+                                        id: null, // to be determined
+                                        uuid: imageUUID,
+                                        appKey: req.appKey,
+                                        tenant: req.tenant,
+                                        sourceFile: sourceFile,
+                                        targetFile: targetFile,
+                                        op: opItem,
+                                    });
+                                    nextOp();
+                                }
+                            });
+                        }
+                        else nextOp();
+                    },
+                    // Finished operations
+                    (err) => {
+                        if (err) next(err);
+                        else next();
+                    }
+                );
+            },
+            
+            // Record images in database
+            (next) => {
+                async.eachSeries(
+                    processedImages,
+                    (image, nextImage) => {
+                        // TODO: adjust query to match the actual DB table structure
+                        this.db.query(
```

Comment:
How about pulling this out into a Model object, and just calling ```Model.insert([ {values} ], cb);``` here?

Reply:
I'm hesitant to do that here since there's only going to be this one single DB call for the entire service. But it should make mocking a little easier, so I will probably try that.